contracts/paymaster/BasePaymaster.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: GPL-3.0-only
+// COPIED FROM: https://github.com/opengsn/gsn/blob/master/packages/contracts/src/BasePaymaster.sol
+// Changes:
+// - Drops Ownable (OpenZeppelin).
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+
+import "@opengsn/contracts/src/interfaces/IPaymaster.sol";
+import "@opengsn/contracts/src/interfaces/IRelayHub.sol";
+import "@opengsn/contracts/src/forwarder/IForwarder.sol";
+import "@opengsn/contracts/src/utils/GsnEip712Library.sol";
+
+/**
+ * @notice An abstract base class to be inherited by a concrete Paymaster.
+ * A subclass must implement:
+ *  - preRelayedCall
+ *  - postRelayedCall
+ */
+abstract contract BasePaymaster is IPaymaster, ERC165 {
+  using ERC165Checker for address;
+
+  IRelayHub internal relayHub;
+  address private _trustedForwarder;
+
+  /// @inheritdoc IPaymaster
+  function getRelayHub() public view override returns (address) {
+    return address(relayHub);
+  }
+
+  //overhead of forwarder verify+signature, plus hub overhead.
+  uint256 public constant FORWARDER_HUB_OVERHEAD = 50000;
+
+  //These parameters are documented in IPaymaster.GasAndDataLimits
+  uint256 public constant PRE_RELAYED_CALL_GAS_LIMIT = 100000;
+  uint256 public constant POST_RELAYED_CALL_GAS_LIMIT = 110000;
+  uint256 public constant PAYMASTER_ACCEPTANCE_BUDGET = PRE_RELAYED_CALL_GAS_LIMIT + FORWARDER_HUB_OVERHEAD;
+  uint256 public constant CALLDATA_SIZE_LIMIT = 10500;
+
+  /// @inheritdoc IERC165
+  function supportsInterface(bytes4 interfaceId) public view virtual override(IERC165, ERC165) returns (bool) {
+    return interfaceId == type(IPaymaster).interfaceId || super.supportsInterface(interfaceId);
```

Comment:
I am not 100% sure what `super` refers to here... Is it `ERC165`? Let's just make sure that whichever parent gets called, it also has `supportsInterface(bytes4)`...